### PR TITLE
ProgressWidget: Add an optional marker on the initial position

### PR DESCRIPTION
**Pull request by NiLuJe**
* Enable it on SkimToWidget
* Optional on ReaderFooter (toggle in the progress bar > style submenu)

### CODE REVIEW

**Comment by NiLuJe** [comment]
Full disclosure: I never read with ReaderFooter displayed, so it's setup as an on-demand status panel. So, my progress bar is twice the height of the default. Which is why I started with a design that sorta worked there.

A problem emerged at smaller sizes, where it became nearly invisible, so I switched to another design. it worked better at small sizes, but became very annoying at larger sizes.

TL;DR: There's a random "it looks good starting from this size" threshold that switches between the two.

"Large" design in SkimTo, and "small" design at its largest size in Footer (12):

v2_top

**Comment by NiLuJe** [comment]
"Small" design on the default sizes, for thin & thick

v2_thin
v2_7

**Comment by NiLuJe** [comment]
"Large" design at my usual size (14):

v2_14

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 5 of 5 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10114)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

**Comment by NiLuJe** [comment]
It's very much inspired by the old Kindle progress bar, yeah (I remember the cries of outrage when it went poof ;p).

Right now, it's set on document open, and that's it. Hadn't thought much about it, but a reset on resume sorta sounds sensible?

**Comment by mergen3107** [comment]
Ok, gotcha! 

I will be happy to test this out. 

If you are into this, maybe let’s make it more flexible?
Like, renew initial position only if suspend / document switch occurred > 1 hour? (configurable?)

**Comment by NiLuJe** [comment]
That sounds like over-engineering for something that should do a very simple thing, and that's it ;o).

The only complication that might solve a real usability problem I can think of would *not* resetting if the suspend didn't last long, but I'd be wary of even trying to deal with that.

**Comment by NiLuJe** [comment]
Note to self: cherry-pick 6f5431f92

(And double-check whether it makes more sense onSuspend or onResume, because I have a vague memory of... something pushign the current choice).
